src/accordion/accordion.directive.ts

```diff
@@ -0,0 +1,305 @@
+		ngbCollapse.collapsed = this.collapsed;
+		// we set the animation to the default of the accordion
+		ngbCollapse.animation = this.accordion.animation;
+		// event forwarding from 'ngbCollapse' to 'ngbAccordion'
+		this.subscriptions.push(
+			ngbCollapse.hidden.subscribe(() => {
+				// when the animation finishes we can remove the template from DOM
+				this.animatingBodyCollapse = false;
+				this.hidden.emit();
+				this.accordion.hidden.emit(this.id);
+			}),
+			ngbCollapse.shown.subscribe(() => {
+				this.shown.emit();
+				this.accordion.shown.emit(this.id);
+			}),
+		);
+	}
+
+	toggle() {
+		this.collapsed = !this.collapsed;
+	}
+}
+
+@Directive({
+	exportAs: 'ngbAccordion',
+	standalone: true,
+	selector: '[ngbAccordion]',
+	host: { '[class.accordion]': 'true' },
+})
+export class NgbAccordionDirective {
+	@ContentChildren(NgbAccordionItem, { descendants: false }) private items: QueryList<NgbAccordionItem>;
+	/**
+	 * If `true`, accordion will be animated.
+	 */
+	@Input() animation: boolean;
+	/**
+	 * If `true`, only one item at the time can stay open.
+	 */
+	@Input() closeOthers: boolean;
+	/**
+	 * If `true`, the content of the AccordionBody will be removed from the DOM.
+	 * This property can be overwritten at AccordionItem level
+	 */
+	@Input() destroyOnHide = true;
+
+	/**
+	 * An event emitted when the expanding animation is finished on the collapse. The payload is the AccordionItem id.
+	 */
+	@Output() shown = new EventEmitter<string>();
+
+	/**
+	 * An event emitted when the collapsing animation is finished on the collapse, and before the content of the AccordionBody is removed.
+	 * The payload is the AccordionItem id.
+	 */
+	@Output() hidden = new EventEmitter<string>();
+
+	constructor(config: NgbAccordionConfig) {
+		this.animation = config.animation;
+		this.closeOthers = config.closeOthers;
+	}
+
+	/**
+	 * Toggles a panel with the given id.
+	 *
+	 * Has no effect if the panel is disabled.
+	 */
+	toggle(itemId: string) {
+		if (this.canBeToggled(itemId)) {
+			this.items.forEach((item) => {
+				if (item.id === itemId) {
+					item.toggle();
+				} else if (this.closeOthers) {
+					item.collapsed = true;
+				}
+			});
+		}
+	}
+
+	/**
+	 * Expands panel with the given id.
+	 *
+	 * Has no effect if the panel is disabled. If `closeOther` is `true` it will collapse the other panels.
+	 */
+	expand(itemId: string) {
+		if (this.canBeToggled(itemId)) {
+			this.items.forEach((item) => {
+				if (item.id === itemId) {
+					item.collapsed = false;
+				} else if (this.closeOthers) {
+					item.collapsed = true;
+				}
+			});
+		}
+	}
+
+	expandAll() {
+		if (this.closeOthers) {
+			const available = this.items.filter((item) => !item.disabled);
+			//we check if there are any item not disabled and none of them is expanded
+			if (available.length > 0 && !available.find((item) => !item.collapsed)) {
+				available[0].collapsed = false;
+			}
+		} else {
+			this.items.forEach((item) => {
+				if (!item.disabled) {
```

Comment:
I'm not sure about `disabled` handling again... why can't you work with `disabled` items programmatically? I understand that the user can't do anything with them, but why do APIs not allow it?